kartothek/io/dask/dataframe.py

```diff
@@ -405,3 +406,61 @@ def collect_dataset_metadata(
         ddf = dd.from_pandas(df, npartitions=1)
 
     return ddf
+
+
+def _unpack_hash(df, unpack_meta):
+    df2 = unpack_payload_pandas(df, unpack_meta)
+    return _hash_partition(df2)
+
+
+def _hash_partition(part):
+    return pd.util.hash_pandas_object(part, index=False).sum().astype("uint64")
+
+
+@default_docs
+def hash_dataset(
+    store: Optional[Callable[[], KeyValueStore]] = None,
+    dataset_uuid: Optional[str] = None,
+    subset=None,
+    group_key=None,
+    table: str = SINGLE_TABLE,
+    predicates: Optional[PredicatesType] = None,
+    factory: Optional[DatasetFactory] = None,
+):
+    """
+    Calculate a partition wise, or group wise, hash of the dataset.
+
+    .. note::
+
+        We do not guarantee the hash values to remain constant accross versions.
```

Comment:
What about same values but different data types (e.g. uint64(1) == int64(1))?

Reply:
can you rephrase the question, please? Are you referring to a guarantee about type stability or what are you referring to here? This remark shall simply indicate that we may (or may not) change the implementation without notice and we do not necessarily recommend to use this implementation for long term storage or similar.

For instance, we may want to change how hashes are combined, e.g. instead of doing `hash(partition).sum()` we may want to go for another hashing of the values, e.g. `hash(*hash(partition))` and I don't want anybody to rely on this implementation detail

Reply:
Sorry for the lack of clarity. I meant to say that we might note here whether 2 datasets with the same values but different dtypes will have the same hash.

Reply:
That's also quite another topic entirely. For the sake of the argument, though

TL;DR

Same value == same hash

different type family -> different value -> different hash

```python
In [1]: import pandas as pd

In [2]: pd.util.hash_pandas_object(pd.DataFrame({"int": [1]})) == pd.util.hash_pandas_object(pd.DataFrame({"int": [1]}).astype({"int": "int8"}))
Out[2]:
0    True
dtype: bool

In [3]: pd.util.hash_pandas_object(pd.DataFrame({"int": [1]})) == pd.util.hash_pandas_object(pd.DataFrame({"int": [1]}).astype({"int": "int32"}))
Out[3]:
0    True
dtype: bool

In [4]: pd.util.hash_pandas_object(pd.DataFrame({"int": [1]})) == pd.util.hash_pandas_object(pd.DataFrame({"int": [1.0]}))
Out[4]:
0    False
dtype: bool
```

Reply:
Thanks for the explanation.
```suggestion
        We do not guarantee the hash values to remain constant accross versions.
        It should not be expected that the hash of the same value in different data types is identical.
```